src/ui/marker.js

```diff
@@ -444,35 +448,48 @@ export default class Marker extends Evented {
         return this;
     }
 
-    _updateOcclusion() {
-        if (!this._occlusionTimer) {
-            this._occlusionTimer = setTimeout(this._onOcclusionTimer.bind(this), 60);
-        }
-    }
+    _evaluateOpacity() {
+        const position = this._pos ? this._pos.sub(this._transformedOffset()) : null;
 
-    _clearOcclusionTimer() {
-        if (this._occlusionTimer) {
-            clearTimeout(this._occlusionTimer);
-            this._occlusionTimer = null;
+        if (!this._withinScreenBounds(position)) {
+            this._fadeTimer = null;
+            return;
         }
-    }
 
-    _onOcclusionTimer() {
-        const tr = this._map.transform;
-        const pos = this._pos ? this._pos.sub(this._transformedOffset()) : null;
-        if (pos && pos.x >= 0 && pos.x < tr.width && pos.y >= 0 && pos.y < tr.height) {
-            // calculate if occluded.
-            const raycastLoc = this._map.unproject(pos);
+        const lngLat = this._map.unproject(position);
+
+        let terrainOccluded = false;
+        if (this._map.getTerrain()) {
             const camera = this._map.getFreeCameraOptions();
             if (camera.position) {
                 const cameraPos = camera.position.toLngLat();
-                const raycastDistance = cameraPos.distanceTo(raycastLoc);
+                const raycastDistance = cameraPos.distanceTo(lngLat);
```

Comment:
Since it gets name-mangled anyway, is a more descriptive name than lngLat possible since these two lines look ostensibly identical but compute two different things?